### PR TITLE
Games: Leaderboard

### PR DESCRIPTION
**Pull request by delmaass**
This PR introduces the Leaderboard.
It is simply the sum of the Scores' value for each user, sorted in descending order.
You can filter the leaderboard by a Game, to display it next to a Game container.

### CODE REVIEW

File: backend/games/tests/test_leaderboard.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    ########
+    # LIST #
+    ########
+
+    def test_if_not_logged_in_then_cannot_list(self):
+        res = self.list()
+        self.assertStatusCode(res, 401)
+
+    def if_logged_in_then_can_list(self):
+        print(self.login("17simple"))
```
Comment (abocquet) [inline backend/games/tests/test_leaderboard.py:53]: This might not be supposed to be here

File: backend/games/tests/test_leaderboard.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+class LeaderboardTestCase(WeakAuthenticationBaseTestCase):
+    fixtures = ["test_authentication.yaml", "games.yaml", "test_scores.yaml"]
+
+    EXPECTED_FIELDS = {"user", "total_score"}
+
+    def endpoint_list(self):
+        return "/games/scores/leaderboard/"
+
+    def list(self):
```
Comment (abocquet) [inline backend/games/tests/test_leaderboard.py:15]: `list` is a very generic term, it would be easier to read with something like `query_list_endpoint`

File: backend/games/tests/test_leaderboard.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+from games.models.scores import Score
+from backend.tests_utils import WeakAuthenticationBaseTestCase
+from django.db.models import Sum
+
+
+class LeaderboardTestCase(WeakAuthenticationBaseTestCase):
```
Comment (abocquet) [inline backend/games/tests/test_leaderboard.py:7]: Should we add a test that tries to submit a negative score?

**Review by abocquet — APPROVED** [review]
Some nits in the tests, otherwise LGTM

File: backend/games/tests/test_leaderboard.py (hunk 1 above)
Comment (delmaass) [reply backend/games/tests/test_leaderboard.py:53]: Are you talking about the print function ?

File: backend/games/tests/test_leaderboard.py (hunk 3 above)
Comment (delmaass) [reply backend/games/tests/test_leaderboard.py:7]: U are right, it should be tested 👍 I can add this test in `test_scores.py`, in another PR.
